arch/generic/crc32_braid_c.c

```diff
@@ -211,6 +207,539 @@ Z_INTERNAL uint32_t PREFIX(crc32_braid)(uint32_t crc, const uint8_t *buf, size_t
         DO1;
     }
 
+    /* Return the CRC, post-conditioned. */
+    return c;
+}
+
+#define bitbuffersizebytes (128 * 1024)
+#define bitbuffersizeqwords (bitbuffersizebytes / sizeof(uint64_t))
+
+uint32_t chorba_118960_nondestructive (uint32_t crc, const uint8_t* input, size_t len) {
+    uint64_t* bitbuffer = calloc(bitbuffersizeqwords, sizeof(uint64_t));
```

Comment:
I think we probably want zcalloc here (mostly for alignment).  It may also make sense to time this using stack backed memory, instead, since it's just 128k and fixed at compile time.

Reply:
Yes for alignment, also the calloc/zalloc is not needed because the first 2 passes write back directly (instead of XOR) so we don't need memory to be empty. Happy to do this with stack too if you're okay with it.